teps/0089-nonfalsifiable-provenance-support.md

```diff
@@ -224,24 +309,298 @@ The entrypointer image should also be able to see that this flag is set and acco
+1. Verify that a TaskRun that's been modified during execution isn't verified by the TektonController
+1. Verify that a TaskRun that's been modified after execution isn't verified by the TektonController
+1. Verify that a TaskRun that's been modified during execution isn't verified by the Chains
+1. Verify that a TaskRun that's been modified after execution isn't verified by the Chains
+
+## Design Details - API changes
+
+At the moment, this TEP does not introduce API changes. It stores the necessary data as part of the termination message of TaskRun pods (as results), and TaskRun signatures are included as part of the [embedded `Status` annotation object in the `TaskRun.Status` field](https://github.com/tektoncd/pipeline/blob/302895b5a1ca45c02a85a9822201643c159fe02c/pkg/apis/pipeline/v1beta1/taskrun_types.go#L111). Examples are as shown in the above sections.
+
+As the feature matures and graduates to beta/GA, embeding signature meatadata into the status object can be considered:
+```
+// TaskRunStatus defines the observed state of TaskRun
+type TaskRunStatus struct {
+	duckv1beta1.Status `json:",inline"`
+
+    // SignatureMetadata would include the necessary information for signing of the fields of the TaskRun status.
+	SignatureMetadata `json:",inline"`
+
+
+	// TaskRunStatusFields inlines the status fields.
+	TaskRunStatusFields `json:",inline"`
+}
+```
+
+## Design Details - Performance Implications
+
+There are several operations that are considered:
+- Signing and Verification operations
+- Creating/Deleting a SPIRE entry
+- Obtaining an SVID
+- Obtaining a Trust Bundle
+
+Performance analysis:
+- Signing and verification operations are local operations and should be negligible (in microseconds).
+- Creating and deleting a SPIRE entry takes about a round trip time (RTT) to the SPIRE server (and RTT from SPIRE server to DataStore - disk or network). This is done once during the initialization of the TaskRun(in 100-500ms).
+- Obtaining an SVID, this takes a RTT to the SPIRE server (and RTT from SPIRE server to KeyManager - disk or network). 
+  - For the TektonController, this is done whenever the certificate expires (usually hour(s) - configurable). Rather negligible.
+  - For each TaskRun pod, the entrypointer obtains the SVID. This should take a RTT with the SPIRE server. However, there is a caveat here, that the SPIRE server needs to recognize that an entry has been created in order to fulfill the SVID request. The consistency of this can take up to a maximum of 15 seconds (and this happens because entries are created just in time (JIT)), and is in the critical path of the entrypointer's signing mechanism.
+- Obtaining a Trust Bundle takes a RTT to the SPIRE server. The SPIRE server should have this ready to serve and only needs to refresh it once in a long while (~hours if not days) with the UpstreamAuthority.
+
+Overall, there is minimal performance impact, with a slight consideration for TaskRun pod cold start-up latency.
```

Comment:
@pxp928 mentioned that we can skip checking for completed task runs as long as they have been verified by chains. Since it doesn't quite provide value anymore. 

I think this probably is the cleanest solution for now.